diff --git a/lib/message/converter.js b/lib/message/converter.js
index c821bb3cdb5a404b1a7ef66883c3b22192b0c16d..e97774395375138be6c4489142ea3db79ba66538 100644
--- a/lib/message/converter.js
+++ b/lib/message/converter.js
@@ -92,7 +92,7 @@ class Converter {
             return;
         }
         if (qq === "all") {
-            var q = 0, flag = 1, display = "全体成员";
+            var q = 0, flag = 1, display = text || "全体成员";
         }
         else {
             var q = Number(qq), flag = 0, display = text || String(qq);
@@ -101,7 +101,6 @@ class Converter {
                 display = member?.card || member?.nickname || display;
             }
         }
-        display = "@" + display;
         if (dummy)
             return this._text(display);
         const buf = Buffer.allocUnsafe(6);
