diff --color -urpN a/lib/node_modules/prisma/build/index.js b/lib/node_modules/prisma/build/index.js
--- a/lib/node_modules/prisma/build/index.js	1970-01-01 08:00:01.000000000 +0800
+++ b/lib/node_modules/prisma/build/index.js	2022-09-11 20:37:17.007631971 +0800
@@ -85476,6 +85476,7 @@ async function gracefulExec(cmd) {
 }
 __name(gracefulExec, "gracefulExec");
 async function getPlatform() {
+  return "linux-nixos";
   const { platform: platform2, libssl, distro, arch } = await getos();
   if (platform2 === "darwin" && arch === "arm64") {
     return "darwin-arm64";
